pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -142,6 +141,47 @@ func (cb *ClusterBuilder) applyDestinationRule(c *cluster.Cluster, clusterMode C
 	return subsetClusters
 }
 
+// MergeTrafficPolicy returns the merged TrafficPolicy for a destination-level and subset-level policy on a given port.
+func MergeTrafficPolicy(original, subsetPolicy *networking.TrafficPolicy, port *model.Port) *networking.TrafficPolicy {
```

Comment:
Can you please add a test for this function?

Reply:
This is basically a copy/paste of the original method `SelectTrafficPolicyComponents` from cluster.go with only minor changes.  Coverage indirectly comes by the tests in cluster_test.go and those tests were good enough to detect issues like the one I mentioned [below](https://github.com/istio/istio/pull/25711#discussion_r458770640).  I suppose moving the method from cluster.go to cluster_builder.go may have impacted codecov.  @ramaraochavali if you want additional testing besides what's in cluster_test.go, let me know and I can add some

Reply:
I am thinking the following test cases would help. Not sure if we have test cases that validate these cases - if not it would be good to add them
- a component in subset traffic policy like load balancer overriding destination rule traffic policy
- overriding port level traffic policy in subsets

Reply:
tests added